core/capabilities/capability.go

```diff
@@ -0,0 +1,145 @@
+package capabilities
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"golang.org/x/mod/semver"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+// CapabilityType is an enum for the type of capability.
+type CapabilityType string
```

Comment:
Why not an int? I imagine we would have a similar enum onchain and there a small int will be cheaper than a string.